app/src/main/java/com/yapp/bol/app/di/RepositoryModule.kt

```diff
@@ -0,0 +1,24 @@
+package com.yapp.bol.app.di
+
+import com.yapp.bol.data.datasource.impl.MockDataSourceImpl
+import com.yapp.bol.data.repository.MockRepositoryImpl
+import com.yapp.bol.domain.repository.MockRepository
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+class RepositoryModule {
+    @Provides
+    @Singleton
+    fun provideMockRepository(
+        mockDataSourceImpl: MockDataSourceImpl,
+    ): MockRepository {
+        return MockRepositoryImpl(
+            mockDataSourceImpl,
+        )
```

Comment:
p5 : 한줄로 처리 가능한 함수는 single Function 으로 만들면 가독성이 좋아집니다. 
```suggestion
    ): MockRepository = MockRepositoryImpl(mockDataSourceImpl)
```
p5 : 한줄로 처리 가능한 함수는 [single-expression-function](https://kotlinlang.org/docs/functions.html#single-expression-functions) 으로 만들면 가독성이 좋아집니다. 
아래 부분은 single-expression-function 으로 만드셨던데, 기준이 있을가요?

Reply:
마찬가지로 @Binds 를 사용하는게 코드 생성을 줄일 수 있을 것 같네요~~

Reply:
두 분 다 좋은 리뷰 감사합니다! 

> 아래 부분은 single-expression-function 으로 만드셨던데, 기준이 있을가요?

single expression function 안 쓴 이유가 어렴풋이 기억나는데.. ㅋㅋㅋ 제 기억에 저기서 다른 코드 찍어보다가.. 삭제는 했는데 미처 single expression function으로 돌릴 생각은 안 했던 것 같아요. 앞으론 좀 꼼꼼하게 처리해볼게요! 

> 마찬가지로 https://github.com/BINDS 를 사용하는게 코드 생성을 줄일 수 있을 것 같네요~~

제가 Hilt 사용이 익숙치 않아서 @Binds를 어디서 사용해야 할지 헤매고 있었는데 이런 거였군요..! 감사합니다!! 주신 링크도 학습하는데 큰 도움이 되었습니다! 👍🙇